packages/devtools_app/lib/src/debugger/debugger_controller.dart

```diff
@@ -75,26 +75,36 @@ class DebuggerController extends DisposableController
 
   VmServiceWrapper _lastService;
 
-  void _handleConnectionAvailable(VmServiceWrapper service) {
+  Future<void> _handleConnectionAvailable(VmServiceWrapper service) async {
     if (service == _lastService) return;
     _lastService = service;
     onServiceShutdown();
     if (service != null) {
-      initialize();
+      await initialize();
     }
   }
 
-  void initialize() {
+  bool get initialized => _initialized;
+
+  bool _initialized = false;
+
+  Future<void> initialize() async {
     if (initialSwitchToIsolate) {
       assert(serviceManager.isolateManager.selectedIsolate.value != null);
-      switchToIsolate(serviceManager.isolateManager.selectedIsolate.value);
+      await switchToIsolate(
+        serviceManager.isolateManager.selectedIsolate.value,
+      );
     }
 
-    addAutoDisposeListener(serviceManager.isolateManager.selectedIsolate, () {
-      switchToIsolate(serviceManager.isolateManager.selectedIsolate.value);
+    addAutoDisposeListener(serviceManager.isolateManager.selectedIsolate,
+        () async {
+      await switchToIsolate(
+        serviceManager.isolateManager.selectedIsolate.value,
+      );
     });
     autoDispose(_service.onDebugEvent.listen(_handleDebugEvent));
     autoDispose(_service.onIsolateEvent.listen(_handleIsolateEvent));
+    _initialized = true;
```

Comment:
what does this accomplish? the isolate won't be selected by the end of this method with the awaits.